test/clone.js

```diff
@@ -189,3 +189,20 @@ describe('deep clone edge cases', function() {
     assert.notStrictEqual(R.clone(list), list);
   });
 });
+
+
+describe('Let `R.clone` use an arbitrary user defined `clone` method', function() {
+
+  it('dispatches to `clone` method if present', function() {
+    function ArbitraryClone(x) { this.value = x; }
+
+    // user defined `clone` method can or cannot make sense: it's arbitrary.
+    ArbitraryClone.prototype.clone = function() { return new ArbitraryClone(2 * this.value); };
```

Comment:
Although clearly an `ArbitraryClone` would be allowed to do whatever it wanted to, I'd rather _not_ call attention to it in the test suite, just doing, `return new ArbitraryClone(this.value)`.

If you find this that important, I think it should just be a comment.  ("Note that if Ramda delegates to the object's 'clone' method, it does no checking that data is similar," or some such.)  But personally, I don't find it really worth spending any effort on.